AppCenter/AppCenter/Internals/Channel/MSOneCollectorChannelDelegate.m

```diff
@@ -96,31 +120,67 @@ - (void)channel:(id<MSChannelProtocol>)channel didSetEnabled:(BOOL)isEnabled and
   }
 }
 
-- (void)channel:(id<MSChannelProtocol>)channel didPrepareLog:(id<MSLog>)log withInternalId:(NSString *)__unused internalId {
-  if (![self shouldSendLogToOneCollector:log] ||
-      ![channel conformsToProtocol:@protocol(MSChannelUnitProtocol)]) {
-    return;
-  }
-  id<MSChannelUnitProtocol> channelUnit = (id<MSChannelUnitProtocol>)channel;
-  NSString *groupId = channelUnit.configuration.groupId;
-  id<MSChannelUnitProtocol> oneCollectorChannelUnit = [self.oneCollectorChannels objectForKey:groupId];
-  if (!oneCollectorChannelUnit) {
-    return;
-  }
-  id<MSLogConversion> logConversion = (id<MSLogConversion>)log;
-  NSArray<MSCommonSchemaLog *> *commonSchemaLogs = [logConversion toCommonSchemaLogs];
-  for (MSCommonSchemaLog *commonSchemaLog in commonSchemaLogs) {
-    [oneCollectorChannelUnit enqueueItem:commonSchemaLog];
-  }
-}
+#pragma mark - Helper
 
 - (BOOL)isOneCollectorGroup:(NSString *)groupId {
   return [groupId hasSuffix:kMSOneCollectorGroupIdSuffix];
 }
 
 - (BOOL)shouldSendLogToOneCollector:(id<MSLog>)log {
   NSObject *logObject = (NSObject *)log;
-  return [[log transmissionTargetTokens] count] > 0 && [log conformsToProtocol:@protocol(MSLogConversion)] && ![logObject isKindOfClass:[MSCommonSchemaLog class]];
+  return [[log transmissionTargetTokens] count] > 0 && [log conformsToProtocol:@protocol(MSLogConversion)] &&
+         ![logObject isKindOfClass:[MSCommonSchemaLog class]];
+}
+
+- (BOOL)validateLog:(MSCommonSchemaLog *)log {
+  if (![self validateLogName:log.name]) {
+    return NO;
+  }
+  if (![self validateLogData:log.data]) {
+    return NO;
+  }
+  return YES;
+}
+
+- (BOOL)validateLogName:(NSString *)name {
+
+  // Name mustn't be nil.
+  if (!name.length) {
+    MSLogError([MSAppCenter logTag], @"%@ Name must not be nil or empty.", kMSBaseErrorMsg);
+    return NO;
+  }
+
+  // The Common Schema event name must conform to a regex.
+  NSError *error = nil;
+  NSRegularExpression *regex =
+      [NSRegularExpression regularExpressionWithPattern:kMSLogNameRegex options:0 error:&error];
+  NSRange range = NSMakeRange(0, name.length);
+  if (!regex) {
```

Comment:
Does this really need to be checked/logged? The regex is a const, so it should always work. Maybe it would be better to have a unit test that validates the regex is valid

Reply:
Makes sens, I can update that.